nextest-runner/src/reporter/structured/libtest.rs

```diff
@@ -349,31 +367,47 @@ impl<'cfg> LibtestReporter<'cfg> {
                 match last_status.result {
                     ExecutionResult::Fail { .. } | ExecutionResult::ExecFail => {
                         test_suite.failed += 1;
-                        let stdout = String::from_utf8_lossy(&last_status.stdout);
-                        let stderr = String::from_utf8_lossy(&last_status.stderr);
-
-                        // TODO: Get the combined stdout and stderr streams, in the order they
-                        // are supposed to be, to accurately replicate libtest's output
-
-                        // TODO: Strip libtest stdout output
-                        // libtest outputs various things when _not_ using the
-                        // unstable json format that we need to strip to emulate
-                        // that json output, eg.
-                        //
-                        // ```
-                        // running <n> tests
-                        // <test output>
-                        // test <name> ... FAILED
-                        // \n\nfailures:\n\nfailures:\n    <name>\n\ntest result: FAILED
-                        // ```
-
-                        write!(
-                            out,
-                            r#","stdout":"{}{}""#,
-                            EscapedString(&stdout),
-                            EscapedString(&stderr)
-                        )
-                        .map_err(fmt_err)?;
+
+                        // Write the output from the test into the `stdout` (even
+                        // though it could contain stderr output as well).
+                        // Unfortunately, to replicate the libtest json output,
+                        // we need to do our own filtering of the output to strip
+                        // out the data emitted by libtest in the human format
+                        write!(out, r#","stdout":""#).map_err(fmt_err)?;
+
+                        let mut in_test_output = false;
+
+                        for line in last_status.output.lines() {
+                            let line = if in_test_output {
+                                let data = line.lossy();
```

Comment:
Looks like this chunk of code and the one immediately below doesn't have tests. Could you add some?